doc/source/developer_guide/dg_col_phys.rst

```diff
@@ -127,60 +129,89 @@ Overall, columnphysics changes in the Icepack model should include the following
+         character(len=*), parameter :: subname = '(icepack_public_interface)'
 
-    * An example of how this might look is
+         if (flag_arg2) then
+            if (.not.present(arg2)) then
+               call icepack_warnings_setabort(.true.,__FILE__,__LINE__)
+               call icepack_warnings_add(subname//' flag_arg2 set but arg2 not passed')
+            endif
+         endif
+         if (flag_arg3) then
+            if (.not.present(arg3)) then
+               call icepack_warnings_setabort(.true.,__FILE__,__LINE__)
+               call icepack_warnings_add(subname//' flag_arg3 set but arg3 not passed')
+            endif
+         endif
+         if (icepack_warnings_aborted(subname)) return
 
-      .. code-block:: fortran
+         ...
+         call some_columnphysics_subroutine(arg1, arg2, arg3, ...)
+         ...
+
+         end subroutine
+
+         !------------
+
+         subroutine some_columnphysics_subroutine(arg1, arg2, arg3, ...)
 
-         subroutine icepack_example_interface(arg1, arg2, ...)
          real (kind=dbl_kind), intent(inout) :: arg1
          real (kind=dbl_kind), optional, dimension(:), intent(inout) :: arg2
-         !
-         real (kind=dbl_kind), allocatable, dimension(:) :: l_arg2
-         logical :: arg2_flag
-
-         character(len=*), parameter :: subname = '(icepack_example_interface)'
-
-         if (present(arg2)) then
-            arg2_flag = .true.
-            allocate(l_arg2(size(arg2)))
-            l_arg2 = arg2
-         else
-            arg2_flag = .false.
-            allocate(l_arg2(1))
-            l_arg2 = c0
+         real (kind=dbl_kind), optional, intent(inout) :: arg3
+
+         if (flag_arg2) then
+            arg2(:) = ...
          endif
 
-         ...
+         if (flag_arg3) then
+            call someother_columnphysics_subroutine(arg3)
+         endif
 
-         call some_columnphysics_subroutine(arg1, l_arg2, arg2_flag, ...)
+         end subroutine
 
-         ...
+         !------------
 
-         if (present(arg2)) then
-            arg2 = l_arg2
-         endif
-         deallocate(l_arg2)
+         subroutine someother_columnphysics_subroutine(arg3)
+
+         real (kind=dbl_kind), optional, intent(inout) :: arg3
+
+         arg3 = ...
 
-         return
          end subroutine
 
+
+    Some notes
+
+    * If optional arguments are passed but not needed, this is NOT an error.
+
+    * If checking and implementation are done properly, optional arguments that are not needed will never be referenced anywhere in Icepack at that timestep
+
+    * There is a unit test in CICE to verify robustness of this approach.
+
+    * We recommend doing all checks for optional arguments for an interface before returning just for completeness (as shown above)
+
+    * An argcheck parameter will control when to do the checks, 'none', 'first', or 'all' may be possible settings 
```

Comment:
Is this argcheck parameter implemented anywhere now?

Reply:
Not yet, but soon.